issues: tweak output template for nicer emails

tbg opened the pull request:
- issues: make artifacts link stand out more
- issues: don't emit internal log when no author provided
- roachtest: don't inline the repro helper script
- issues: tweak the footer

tbg committed:
issues: make artifacts link stand out more

Release note: None

tbg committed:
issues: don't emit internal log when no author provided

Roachtests intentionally don't provide authors, so it's annoying
to have the "internal log" box pop up in every issue filed.

Release note: None

tbg committed:
roachtest: don't inline the repro helper script

It takes too much space in emails (which don't render the
details/summary HTML).

Release note: None

tbg committed:
issues: tweak the footer

[Final result]

[Final result]: https://github.com/cockroachdb/cockroach/issues/new?body=storage.kv%2Fsplits%2Fnodes%3D3%2Fquiesce%3Dtrue+%5Bfailed%5D%28https%3A%2F%2Fteamcity.example.com%2FviewLog.html%3FbuildId%3D8008135%26tab%3DbuildLog%29+with+%5Bartifacts%5D%28https%3A%2F%2Fteamcity.example.com%2FviewLog.html%3FbuildId%3D8008135%26tab%3Dartifacts%23%2Fkv%2Fsplits%2Fnodes%3D3%2Fquiesce%3Dtrue%29+on+release-0.1+%40+%5Babcd123%5D%28https%3A%2F%2Fgithub.com%2Fcockroachdb%2Fcockroach%2Fcommits%2Fabcd123%29%3A%0A%0A%0A%60%60%60%0AThe+test+failed+on+branch%3Dmaster%2C+cloud%3Dgce%3A%0A%60%60%60%0A%3Cdetails%3E%3Csummary%3EReproduce%3C%2Fsummary%3E%0A%3Cp%3E%0A%3Cp%3EParameters+in+this+failure%3A%0A%0A-+TAGS%3Ddeadlock%0A%0A-+GOFLAGS%3Drace%0A%3C%2Fp%3E%0A%3C%2Fp%3E%0A%3C%2Fdetails%3E%0A%3Cdetails%3E%3Csummary%3ESame+failure+on+other+branches%3C%2Fsummary%3E%0A%3Cp%3E%0A%0A-+%2331+boom+related+%5BC-test-failure+O-robot+release-0.2%5D%0A%3C%2Fp%3E%0A%3C%2Fdetails%3E%0A%3Csub%3E%0A%0A%5BThis+test+on+roachdash%5D%28https%3A%2F%2Froachdash.crdb.dev%2F%3Ffilter%3Dstatus%3Aopen%2520t%3A.%2Akv%2Fsplits%2Fnodes%3D3%2Fquiesce%3Dtrue.%2A%26sort%3Dtitle%2Bcreated%26display%3Dlastcommented%2Bproject%29+%7C+%5BImprove+this+report%21%5D%28https%3A%2F%2Fgithub.com%2Fcockroachdb%2Fcockroach%2Ftree%2Fmaster%2Fpkg%2Fcmd%2Finternal%2Fissues%29%0A%3C%2Fsub%3E%0A&title=storage%3A+kv%2Fsplits%2Fnodes%3D3%2Fquiesce%3Dtrue+failed

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63385)
<!-- Reviewable:end -->

tbg committed:
issues: don't assign author, mention them

We've wanted to move off assigning authors for a while. It turns out
that I also just [broke] something about the assignments, so it's a
convenient time to make the switch.

We still ping the author on the issue, which is going to stay that
way until we've put the plumbing in place that will allow mentioning
and assigning into the project of teams from unit tests too (to date
this is only done from roachtest).

[broke]: https://github.com/cockroachdb/cockroach/issues/63393#issuecomment-816885919

Release note: None

erikgrinaker reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 17 of 17 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 17 of 17 files at r4, 18 of 18 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63385#-:-MY4PN5UDzKEfwKyZYt2:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan and @tbg)

___
*[pkg/cmd/internal/issues/formatter_unit.go, line 114 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/63385#-MY4OX279jeVffI8aq7M:-MY4OX279jeVffI8aq7N:b-lkvvh8) ([raw file](https://github.com/cockroachdb/cockroach/blob/478c5df3c75575ff7d749721b8dc5f6b0a233a31/pkg/cmd/internal/issues/formatter_unit.go#L114)):*
> ```Go
> 
> 		r.HTML("sub", func() {
> 			r.Escaped("\n\n") // need blank like to <sub> tag for whatever reason
> ```

s/like/line/


<!-- Sent from Reviewable.io -->

tbg commented:
TFTR!

bors r=erikgrinaker
